Sharlayan/Reader.Actor.cs

```diff
@@ -75,25 +75,30 @@ public static partial class Reader {
                 }
 
                 foreach (KeyValuePair<uint, ActorItem> kvp in MonsterWorkerDelegate.ActorItems) {
-                    result.RemovedMonsters.TryAdd(kvp.Key, JsonUtilities.Clone(kvp.Value));
+                    result.RemovedMonsters.TryAdd(kvp.Key, kvp.Value.Clone());
                 }
 
                 foreach (KeyValuePair<uint, ActorItem> kvp in NPCWorkerDelegate.ActorItems) {
-                    result.RemovedNPCs.TryAdd(kvp.Key, JsonUtilities.Clone(kvp.Value));
+                    result.RemovedNPCs.TryAdd(kvp.Key, kvp.Value.Clone());
                 }
 
                 foreach (KeyValuePair<uint, ActorItem> kvp in PCWorkerDelegate.ActorItems) {
-                    result.RemovedPCs.TryAdd(kvp.Key, JsonUtilities.Clone(kvp.Value));
+                    result.RemovedPCs.TryAdd(kvp.Key, kvp.Value.Clone());
                 }
 
                 foreach (KeyValuePair<IntPtr, IntPtr> kvp in uniqueAddresses) {
                     try {
-                        byte[] source = MemoryHandler.Instance.GetByteArray(new IntPtr(kvp.Value.ToInt64()), sourceSize);
+                        var characterAddress = new IntPtr(kvp.Value.ToInt64());
+                        byte[] source = MemoryHandler.Instance.GetByteArray(characterAddress, sourceSize);
 
                         // var source = MemoryHandler.Instance.GetByteArray(characterAddress, 0x3F40);
                         var ID = BitConverter.TryToUInt32(source, MemoryHandler.Instance.Structures.ActorItem.ID);
                         var NPCID2 = BitConverter.TryToUInt32(source, MemoryHandler.Instance.Structures.ActorItem.NPCID2);
-                        var Type = (Actor.Type) source[MemoryHandler.Instance.Structures.ActorItem.Type];
+                        var Type = Actor.Type.Unknown;
+                        if (Enum.TryParse(source[MemoryHandler.Instance.Structures.ActorItem.Type].ToString(), out Actor.Type type)) {
```

Comment:
For efficiency, I'd recommend:
```
if (Enum.IsDefined(typeof(Actor.Type), source[MemoryHandler.Instance.Structures.ActorItem.Type]))
{
    Type = (Actor.Type) source[MemoryHandler.Instance.Structures.ActorItem.Type];
}
```
Since it doesn't need to do costly string conversion/parsing.